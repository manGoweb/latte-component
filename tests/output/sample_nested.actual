before{capture $capture195845}in-alpha before<beta>in-beta</beta>in-alpha after{/capture}{include componentsDir/alpha.latte, children=$capture195845}>after