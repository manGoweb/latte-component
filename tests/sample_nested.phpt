<?php declare(strict_types=1);

require __DIR__ . '/bootstrap.php';

$template = /** @lang Latte */ <<<'EOC'
before
<alpha>
	in-alpha before
	<beta>in-beta</beta>
	in-alpha after
</alpha>
after
EOC;

$expected = /** @lang Latte */ <<<'EOC'
before
{capture $capture776139}
	in-alpha before
	{capture $capture195845}
		in-beta
	{/capture}
	{include componentsDir/beta.latte, children=>$capture195845}
	in-alpha after
{/capture}
{include componentsDir/alpha.latte, children=>$capture776139}
after
EOC;

$parser = new \Latte\Parser();
$engine = new \Mangoweb\LatteComponents\Engine($parser);
$engine->registerComponent(new \Mangoweb\LatteComponents\ComponentDefinition('alpha', 'componentsDir/alpha.latte'));
$engine->registerComponent(new \Mangoweb\LatteComponents\ComponentDefinition('beta', 'componentsDir/beta.latte'));
assertRenders($expected, $template);
