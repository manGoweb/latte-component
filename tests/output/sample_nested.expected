before{capture $capture776139}in-alpha before{capture $capture195845}in-beta{/capture}{include componentsDir/beta.latte, children=>$capture195845}in-alpha after{/capture}{include componentsDir/alpha.latte, children=>$capture776139}after